### ISSUE TITLE
Error in httpuv::startServer

### ISSUE THREAD

**Issue by sckott** [rstudio/httpuv#34]
This example works fine for me https://gist.github.com/sckott/20c49a65162dda1a6a83 - but another person on a Linux box is getting the error with the line

``` r
oauth2.0_token(oauth_endpoints("github"), myapp)
```

``` r
Error in httpuv::startServer("127.0.0.1", 1410, list(call = listen)) : 
  Failed to create server
```

I'm not sure if this is a Linux specific problem, or something else.

**Comment by isomorphisms** [comment]
thanks @jcheng5

**Comment by andrewm4894** [comment]
Hey - i'm running into issues here with packages like googlesheets and rdfp that seem to use httrpuv for the oauth. 
 
Here is some output from some some commands:

```
> sessionInfo()
R version 3.4.1 (2017-06-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.2 LTS

Matrix products: default
BLAS: /usr/lib/libblas/libblas.so.3.6.0
LAPACK: /usr/lib/lapack/liblapack.so.3.6.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.4.1
```
And:
```
> shiny::runExample("01_hello")
Loading required package: shiny

Listening on http://127.0.0.1:6838
/usr/bin/xdg-open: 771: /usr/bin/xdg-open: www-browser: not found
/usr/bin/xdg-open: 771: /usr/bin/xdg-open: links2: not found
/usr/bin/xdg-open: 771: /usr/bin/xdg-open: elinks: not found
/usr/bin/xdg-open: 771: /usr/bin/xdg-open: links: not found
/usr/bin/xdg-open: 771: /usr/bin/xdg-open: lynx: not found
/usr/bin/xdg-open: 771: /usr/bin/xdg-open: w3m: not found
xdg-open: no method available for opening 'http://127.0.0.1:6838'

```
And:
```
> shiny::runExample("01_hello", port=1410)
Loading required package: shiny

Listening on http://127.0.0.1:1410
Error in startServer(host, port, handlerManager$createHttpuvApp()) : 
  Failed to create server
```

Does this just mean that the port is blocked - is there any way to know what port is going to be used before i make the call and kill that pid in advance (feels ugly). 

Or do you see anything above that might suggest something else is going on?

Cheers - am kinda struggling with this one given a lot of what is going on is abstracted so far away from where i call it. 

My real problem is that when i run:

```
> library(googlesheets)
> gs_auth()
Error in httpuv::startServer(use$host, use$port, list(call = listen)) : 
  Failed to create server
> 
```
Was trying to use the shiny example as a reproducible error.
